app/js/controller/list_controller.js

```diff
@@ -47,18 +51,38 @@ export default class ListController extends Controller {
     document.body.appendChild(this.detailsView.el);
     this.alertDialog = document.body.querySelector('#alert-dialog');
 
-    this.list = this.model.getAppList();
-    this.appView.update(this.list);
-    this.addonView.update(this.list);
     this.appView.onInstall(this.handleInstall.bind(this));
     this.addonView.onInstall(this.handleInstall.bind(this));
     this.appView.onDetails(this.handleDetails.bind(this));
     this.addonView.onDetails(this.handleDetails.bind(this));
     this.detailsView.onClose(this.handleCloseDetails.bind(this));
     this.detailsView.onInstall(this.handleInstall.bind(this));
-    this.refreshInstalledList();
 
-    this.alreadyCreated = true;
+    this.getApps();
+    this.watchConnection();
+  }
+
+  getApps() {
+    this.initialized = this.model.getAppList().then(list => {
```

Comment:
This seems weird setting `this.initialized` to a Promise and then using it like a boolean above. Let's just do:
```
this.initialized = true;
this.mode.getAppList().then(.....
```